### PR TITLE
Send requests asynchronously

### PR DESCRIPTION
**Pull request by lukasmatusiewicz**
- Suit some main functions to Task-based Async Pattern. This will allow requests to be sent asynchronously or even canceled while the operation is still in progress (optional use).
- Name private fields up to the convention.
- Include nullability.
- Remove redundant files from the repo and add them to .gitignore.
- Code formatting.

### CODE REVIEW

**Comment by lukasmatusiewicz** [comment]
@nilsbehlen , should this commented code be removed, or fixed, or should stay as it is?
https://github.com/privacyidea/csharp-client/blob/main/PrivacyIDEA_Client/PrivacyIDEA.cs#L327

File: PrivacyIDEA_Client/PrivacyIDEA.cs [hunk 1]
```diff
@@ -286,35 +293,36 @@ public PIResponse ValidateCheck(string user, string otp, string? transactionid =
...
             }
 
-                var map = new Dictionary<string, string>
-                    {
-                        { "username", serviceuser },
-                        { "password", servicepass }
-                    };
+            var parameters = new Dictionary<string, string>
+                {
+                    { "username", _Serviceuser }, // ServiceAccountAvailable already passed
```
Comment (nilsbehlen) [inline PrivacyIDEA_Client/PrivacyIDEA.cs:316]: i do not think the comment is necessary here

**Review by nilsbehlen — CHANGES_REQUESTED** [review]
we should probably merge #4 then look at this again since there is some overlap -> merge conflicts

File: PrivacyIDEA_Client/PrivacyIDEA.cs (hunk 1 above)
Comment (lukasmatusiewicz) [reply PrivacyIDEA_Client/PrivacyIDEA.cs:316]: Is removed in #4

**Comment by lukasmatusiewicz** [comment]
Ready to review:)

**Review by nilsbehlen — APPROVED** [review]

**Pull request merged**